File: src/test/java/com/fractalwrench/bpp/common/StringFileReaderTest.java
package com.fractalwrench.bpp.common;

import org.junit.Test;

import java.io.File;
import java.net.URL;

import static org.junit.Assert.assertEquals;

public class StringFileReaderTest {

    @Test
    public void readFileContents() throws Exception {
        URL resource = getClass().getClassLoader().getResource("input.txt");
        String file = resource.getFile();
        String fileContents = new StringFileReader().readFileContents(new File(file));
        assertEquals("Test!", fileContents);
    }

    @Test(expected = RuntimeException.class)
    public void checkNoFile() throws Exception {
        new StringFileReader().readFileContents(new File("fake.txt"));
    }
}

File: src/test/java/com/fractalwrench/bpp/StringFileReaderTest.kt
package com.fractalwrench.bpp

import org.junit.Test

import java.io.File
import java.net.URL

import org.junit.Assert.assertEquals

class StringFileReaderTest {

    @Test
    @Throws(Exception::class)
    fun readFileContents() {
        val resource = javaClass.classLoader.getResource("input.txt")
        val file = resource!!.file
        val fileContents = StringFileReader().readFileContents(File(file))
        assertEquals("Test!\n", fileContents)
    }

    @Test(expected = RuntimeException::class)
    @Throws(Exception::class)
    fun checkNoFile() {
        StringFileReader().readFileContents(File("fake.txt"))
    }
}